python-lib/plugin_partitioning_utils.py

```diff
@@ -0,0 +1,144 @@
+# -*- coding: utf-8 -*-
+"""Module to get paths of necessary input/output folders partitions"""
+
+import dataiku
+import itertools
+from typing import List, AnyStr, Dict, Generator
+
+from utils import time_logging
+
+DEFAULT_NR_TIME_COMPONENTS = 4
+DEFAULT_INPUT_PATH_REPLACEMENT_DICT = {"%Y": None, "%M": None, "%D": None, "%H": None}
+DEFAULT_OUTPUT_PATH_REPLACEMENT_DICT = {
+    "%Y": "DKU_DST_YEAR",
+    "%M": "DKU_DST_MONTH",
+    "%D": "DKU_DST_DAY",
+    "%H": "DKU_DST_HOUR",
+}
+
+
+@time_logging(log_message="Getting output partition path")
+def get_output_partition_path(folder_id: AnyStr) -> AnyStr:
+    """Function returning the path to the output folder's target partition.
+    If output folder isn't partitioned, returns an empty string
+    Args:
+        folder_id: output folder's id
+    Returns:
+        String containing the output partition's path
+    """
+    # Create folder handle
+    client = dataiku.api_client()
+    folder = client.get_project(dataiku.default_project_key()).get_managed_folder(folder_id)
+
+    # Check partitioning
+    if not folder.get_definition().get("partitioning"):
+        return ""
+
+    # Get partitioning file path pattern
+    file_path_pattern = folder.get_definition().get("partitioning").get("filePathPattern")
+
+    # Get discrete partitioning dimensions
+    dimensions = folder.get_definition().get("partitioning").get("dimensions")
+    discrete_dimensions = [dimension.get("name") for dimension in dimensions if dimension.get("type") != "time"]
+
+    # Generate path placeholders replacements dictionnary
+    path_replacement_dict = DEFAULT_OUTPUT_PATH_REPLACEMENT_DICT.copy()
+    for dimension in discrete_dimensions:
+        replacement_key = f"%{{{dimension}}}"
+        # Get flow variable containing dimension's value
+        replacement_value = dataiku.get_flow_variables()["DKU_DST_" + dimension]
+        path_replacement_dict[replacement_key] = replacement_value
+
+    # Generate path
+    dst_file_path = file_path_pattern
+    for dimension_placeholder in path_replacement_dict.keys():
+        dst_file_path = dst_file_path.replace(dimension_placeholder, path_replacement_dict[dimension_placeholder])
+
+    dst_file_path = "".join(dst_file_path.rsplit(".*"))
+
+    return dst_file_path
+
+
+@time_logging(log_message="Getting input partitions paths")
+def get_input_partitions_paths(folder_id: AnyStr) -> List[AnyStr]:
```

Comment:
you can merge the 2 functions get_input_partitions_paths and get_output_partition_path and make one with an argument like is_input=True/False
Also, I'm not sure to understand why get_input_partitions_paths returns a list of strings and the other just a string ?